components/ILIAS/Forum/classes/Moderation/class.ForumModeratorsTable.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+use ILIAS\Data;
+use ILIAS\UI;
+use Psr\Http\Message\ServerRequestInterface;
+
+class ForumModeratorsTable
+{
+    private ilForumModerators $forum_moderators;
+    protected \ilCtrl $ctrl;
+    protected \ilLanguage $lng;
+    protected UI\Factory $ui_factory;
+    protected ServerRequestInterface $request;
+    protected Data\Factory $data_factory;
+
+    public function __construct(ilForumModerators $forum_moderators)
+    {
+        global $DIC;
+
+        $this->ctrl = $DIC->ctrl();
+        $this->lng = $DIC->language();
+        $this->ui_factory = $DIC->ui()->factory();
+        $this->request = $DIC->http()->request();
+        $this->data_factory = new Data\Factory();
+        $this->forum_moderators = $forum_moderators;
```

Comment:
Can we try to pass as many dependencies as possible in the constructor instead of user `$DIC`?